dashboard/lib/services/partial_registration/user_builder.rb

```diff
@@ -11,6 +11,10 @@ def initialize(request:)
 
       def call
         user = ::User.new_with_session(user_params, session)
+
+        sti_class = ::User.find_sti_class(user_params[:user_type])
+        user = user.becomes!(sti_class) if sti_class
```

Comment:
`User#become!` reinitializes the user instance, which breaks the tricky argument setup done by `User.new_with_session`:
```suggestion
        params = user_params

        user_class = ::User.find_sti_class(params[:user_type]) || ::User
        user = user_class.new_with_session(params, session)
```

Reply:
Ah yes, this is the fix! Thanks @artem-vavilov!